04-spatial-operations.Rmd

```diff
@@ -209,21 +324,60 @@ plot(joined["population_millions"], add = TRUE)
+### Aggregating or dissolving 
+In the subsequent sections, we will present spatial operations that also act on and modify the underlying geometry, namely aggregating (dissolving) and clipping operations.
 
 Like attribute data aggregation, covered in section \@ref(vector-attribute-aggregation), spatial data aggregation (also known as dissolving polygons) can be a way of *condensing* data.
 Aggregated data show some statistic about a variable (typically mean average or total) in relation to some kind of *grouping variable*. 
 For attribute data aggregation the grouping variable is another variable, typically one with few unique values relative to the number of rows.
-The `continent` variable in the `world` dataset is a good example:
-there are only eight unique continents but 177 countries.
-In section \@ref(vector-attribute-aggregation) the aggregation process condensed the `world` dataset down into only eight rows and an aggregated `pop` variable representing the total population per continent (see Figure \@ref(fig:continent-pop)).
-<!-- CAREFUL, in fact, fig:continent-pop has already been a spatial operation (dissolving polygons). You should have shown the table only in the previous section --> 
-Spatial data aggregation is the same conceptually but uses a *spatial* grouping object:
-the *output* is the same, in terms of number of rows/features and geometry, as the *grouping object*, but with new variables corresponding to the input dataset.
-As with spatial subsetting, spatial aggregation operations work by extending existing functions.
-Since mid-2017 (with the release of **sf** `0.5-3`) the base R function `aggregate()` works with a spatial object as a grouping variable.
+The `REGION` variable in the `us_states` dataset is a good example:
+there are only four subregions but 49 states (excluding Hawaii and Alaska).
+In section \@ref(vector-attribute-aggregation) we have already seen how attribute aggregation process condensed the `world` dataset down into only eight rows.
+Spatial data aggregation is the same conceptually but in addition to aggregating the attribute data, it dissolves the underlying polygons.
+Here, we want to aggregate the state population into regions.
+This means that we not only end up with four rows but also with four polygons (out of 49 in the beginning).
+As with spatial subsetting, spatial aggregation operations work by extending existing functions:
+
+```{r}
+regions = aggregate(x = us_states[, "total_pop_15"], by = list(us_states$REGION),
+                    FUN = sum, na.rm = TRUE)
+par(mfrow = c(1, 2))
+plot(us_states[, "total_pop_15"], main = "US states")
+plot(regions[, "total_pop_15"], main = "US regions")
+```
+
+Of course, there is also spatial tidyverse counterpart.
+You can achieve the same with:
+
+```{r, eval = FALSE}
+group_by(us_states, REGION) %>%
+  summarize(sum(pop = total_pop_15, na.rm = TRUE))
+```
+
+<!--Not sure how to elegantly include your circle buffer intersection example -->
```

Comment:
Yep, I was not really sure how to elegantly include the following example. But I guess, if you shortened it substantially, it could be a good transition towards your congruence subsection. However, one has to be a bit careful not to confuse the reader so one has to make clear that this example shows an aggregation without dissolving anything. Maybe a nice idea, because we did so in c3 with the world dataset as well.